<launch>
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <arg name="robot_name" default="sparkal1"/>

    <arg name="labelspace_name" default="ade20k_full"/>
    <arg name="semantic_map_dir" default="$(find semantic_inference)/config/colors"/>
    <arg name="semantic_map_file" default="$(arg labelspace_name).csv"/>
    <arg name="semantic_map_path" default="$(arg semantic_map_dir)/$(arg semantic_map_file)"/>

    <arg name="use_image_interface" default="true"/>
    <arg name="publish_pointcloud" default="true"/>
    <arg name="use_compressed_transport" default="true"/>
    <arg name="rgb_topic" default="/$(arg robot_name)/forward/color/image_raw"/>
    <arg name="rgb_info_topic" default="/$(arg robot_name)/forward/color/camera_info"/>
    <arg name="depth_topic" default="/$(arg robot_name)/forward/depth_registered/image_rect"/>
    <arg name="label_topic" default="/semantic/image_raw"/>
    <arg name="pointcloud_topic" default="$(arg robot_name)/semantic_pointcloud"/>

    <arg name="use_gt_frame" default="true"/>
    <arg name="robot_frame" default="$(arg robot_name)/lamp" if="$(arg use_gt_frame)"/>
    <arg name="robot_frame" default="base_link" unless="$(arg use_gt_frame)"/>
    <arg name="sensor_frame" default="$(arg robot_name)/forward_color_optical_frame"/>
    <arg name="visualizer_frame" default="$(arg robot_name)/map"/>

    <arg name="run_name"/>
    <arg name="dataset_path"/>
    <arg name="dataset_info_file" value="$(arg dataset_path)/$(arg robot_name)/launch/$(arg run_name)_info.launch"/>
    <include file="$(arg dataset_info_file)" pass_all_args="true">
        <arg name="label_topic" value="/oneformer/labels/image_raw"/>
    </include>

    <include file="$(find hydra_ros)/launch/hydra.launch" pass_all_args="true">
        <arg name="extrinsics_mode" value="USE_TF"/>
        <arg name="use_oriented_bounding_boxes" default="true"/>
        <arg name="config_dir" default="$(find hydra)/config/d455_outdoor"/>
        <arg name="rviz_file" default="$(arg robot_name).rviz"/>
        <arg name="rgb_image_transport" value="compressed" if="$(arg use_compressed_transport)"/>
        <arg name="rgb_image_transport" value="raw" unless="$(arg use_compressed_transport)"/>
    </include>
</launch>
